src/components/rooms.js

```diff
@@ -408,3 +426,92 @@ function mapStateToProps(state) {
 }
 
 export const RoomList = connect(mapStateToProps)(FilterableRoomList);
+
+const DeleteRoomDialog = ({ open, loading, onClose, onSend }) => {
+  const translate = useTranslate();
+
+  const DeleteRoomToolbar = props => {
+    return (
+      <Toolbar {...props}>
+        <SaveButton
+          label="resources.rooms.action.erase.title"
+          icon={<DeleteIcon />}
+        />
+        <Button label="ra.action.cancel" onClick={onClose}>
+          <IconCancel />
+        </Button>
+      </Toolbar>
+    );
+  };
+
+  return (
+    <Dialog open={open} onClose={onClose} loading={loading}>
+      <DialogTitle>
+        {translate("resources.rooms.action.erase.title")}
+      </DialogTitle>
+      <DialogContent>
+        <DialogContentText>
+          {translate("resources.rooms.action.erase.content")}
+        </DialogContentText>
+        <SimpleForm
+          toolbar={<DeleteRoomToolbar />}
+          submitOnEnter={false}
+          redirect={false}
+          save={onSend}
+        >
+          <BooleanInput
+            fullWidth
+            source="block"
+            label="resources.rooms.action.erase.fields.block"
+            defaultValue={true}
+          />
+        </SimpleForm>
+      </DialogContent>
+    </Dialog>
+  );
+};
+
+export const DeleteRoomButton = props => {
+  const classes = useStyles(props);
```

Comment:
Please extract the DeleteRoomButton together with the DeleteRoomDialog into a separate file, so it can be reused by other views.